components/builder-api-client/src/error.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2016-2017 Chef Software Inc. and/or applicable contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::error;
+use std::fmt;
+use std::io;
+use std::result;
+
+use hyper;
+use serde_json;
+use url;
+
+use hab_http;
+
+#[derive(Debug)]
+pub enum Error {
+    APIError(hyper::status::StatusCode, String),
+    HabitatHttpClient(hab_http::Error),
+    HyperError(hyper::error::Error),
+    IO(io::Error),
+    JSON(serde_json::Error),
+    URL(url::ParseError),
+}
+
+pub type Result<T> = result::Result<T, Error>;
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let msg = match *self {
+            Error::APIError(ref c, ref m) if m.len() > 0 => format!("[{}] {}", c, m),
+            Error::APIError(ref c, _) => format!("[{}]", c),
+            Error::HabitatHttpClient(ref e) => format!("{}", e),
+            Error::HyperError(ref err) => format!("{}", err),
+            Error::IO(ref e) => format!("{}", e),
+            Error::JSON(ref e) => format!("{}", e),
+            Error::URL(ref e) => format!("{}", e),
+        };
+        write!(f, "{}", msg)
+    }
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        match *self {
+            Error::APIError(_, _) => "Received a non-2XX response code from API",
```

Comment:
In an effort to be consistent, I copied this line from the `builder-depot-client` crate.  So yeah, I think we are.